src/Progress/CircularProgress.tsx

```diff
@@ -0,0 +1,116 @@
+import { FC, ReactNode, useRef, useEffect } from 'react';
+import { ThemeProvider } from '@emotion/react';
+import customTheme from 'src/styles/theme';
+import { arc as d3arc, pie as d3pie } from 'd3-shape';
+import { event, select } from 'd3-selection';
+import { CircularProgressStyle, CenterTextStyle } from './circularProgressStyles';
+export interface CircularProgressProps {
+    /** 
+     * This is to set the width of the pie chart. 
+     * */
+    width: number;
+    /** 
+     * This is to set the text that would appear inside the pie chart. 
+     * */
+    text?: ReactNode;
+    /**
+     * The data that is passed to the Circular Progress Component to render different arcs for each value provided.
+     */
+    data: {
+        label: string;
+        value: number;
+        tooltip?: boolean;
+        key?: string;
+    }[];
+    /**
+     * This is to render the background color for each label that is being passed.
+     */
+    theme: {
+        [key: string]: {
+            bgColor: string;
+            fgColor: string;
+        };
+    };
+}
+
+/**
+ * Circular Progress Component
+ * 
+ * The Circular Progress component is used to render the arcs depending on the value provided for each label.
+ */
+const CircularProgress: FC<CircularProgressProps> = ({ width, text, data, theme }) => {
+
+    const svgRef = useRef<SVGSVGElement>(null);
+    const htmlRef = useRef<HTMLDivElement>(null);
+    const radius = width / 2;
+    const circleRadius = radius - 5;
```

Comment:
is there a minimum size? is it an issue if this is a negative number?

Reply:
I have added code to check if this is negative then it would convert it to positive number. Thanks for pointing it out.